File: core/src/main/java/cloud/vinh/rebirthsaga/game/ecs/components/GridPosition.java
package cloud.vinh.rebirthsaga.game.ecs.components;

import com.artemis.Component;

/** Authoritative grid cell of an entity on the current floor, in floor
 * coordinates (y-up). artemis components need a public constructor. */
public class GridPosition extends Component {
    public int x;
    public int y;

    public GridPosition() {
    }

    public GridPosition(int x, int y) {
        this.x = x;
        this.y = y;
    }
}

File: core/src/main/kotlin/cloud/vinh/rebirthdungeon/game/ecs/components/GridPosition.kt
package cloud.vinh.rebirthdungeon.game.ecs.components

import com.artemis.Component

/** Authoritative grid cell of an entity on the current floor, in floor
 * coordinates (y-up). artemis components need a public constructor; the
 * defaulted parameters also generate the no-arg constructor artemis uses
 * reflectively. */
class GridPosition(var x: Int = 0, var y: Int = 0) : Component()
